src/framework/core/js/ModelTransformationTransforms.js

```diff
@@ -244,7 +244,7 @@ var fluid = fluid || fluid_2_0_0;
         if (!transformSpec.match) {
             fluid.fail("valueMapper requires an array or hash of matches at path named \"match\", supplied ", transformSpec);
         }
-        var value = fluid.model.transform.getValue(transformSpec.defaultInputPath, undefined, transformer);
+        var value = fluid.model.transform.getValue(transformSpec.defaultInputPath, transformSpec.input, transformer);
```

Comment:
I filed a new JIRA ( https://issues.fluidproject.org/browse/FLUID-6179 ), but you will probably want to flesh it out. Regarding the removal time, we could remove it at Infusion 3.0, but it would be more user friendly if we had at least a point release with it deprecated.